addons/point_of_sale/models/pos_config.py

```diff
@@ -410,6 +410,10 @@ def _check_groups_implied(self):
                     field_groups = self.env['res.groups'].concat(*(self.env.ref(it) for it in field_group_xmlids))
                     field_groups.write({'implied_ids': [(4, self.env.ref(field.implied_group).id)]})
 
+    def _check_journal(self):
+        ctx = dict(self.env.context, company_id=self.company_id.id)
+        if not self.env['account.journal'].with_context(ctx).search(['|', ('journal_user', '=', True), ('type', '=', 'cash')]):
```

Comment:
shouldn't the current pos be included in the search?

Reply:
@mart-e There is no direct relation between pos.config and account.journal so we can not put it in current search domain.

Reply:
Actually this is what the `journal_ids` field should be about, right? Why not just make sure there is a value set for this field (e.g. required in the view)

Reply:
@mart-e This is what session create does: https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/models/pos_session.py#L184&L192

If there is no journal_ids set then it will find type cash or journal_user=True journals and set on pos.config, so maybe checking journal_ids field has value or not will not work properly.

Reply:
Thanks for the info. Then, it is there you should make the check. If still no journals are found here you probably have an issue

https://github.com/odoo/odoo/blob/95ceedb7fae707582b881873f7b3230878e76626/addons/point_of_sale/models/pos_session.py#L190

Reply:
👍  we can add check here, Thanks @mart-e